### PR TITLE
Add step_wait, step_wait_func, and step_wait_func_done functions

### PR DESCRIPTION
**Pull request by jgraham**
These allows tests that would normally use set_timeout to wait on a condition
function instead. That means faster execution when the timeout is too long
and more leeway for cases where the timeout would have been marginal or
too short.

### CODE REVIEW

File: resources/testharness.js [hunk 1]
```diff
@@ -2035,6 +2035,74 @@ policies and contribution forms [3].
...
+         *                         timeout_multiplier
+         * @param {string} description Error message to add to assert in case of
+         *                             failure.
+         * @param {number} interval Polling interval in ms
+         * @returns {Promise} Promise resolved once cond is met.
+         *
+         **/
+
+        return new Promise(resolve => {
+            wait_for_callback(cond, resolve, description, timeout, interval);
```
Comment (annevk) [inline resources/testharness.js:2104]:
```suggestion
            this.wait_for_callback(cond, resolve, description, timeout, interval);
```

File: resources/testharness.js [hunk 2]
```diff
@@ -2035,6 +2035,74 @@ policies and contribution forms [3].
...
+         * @param {Function} callback A function taking no arguments to call once
+         *                            the condition is met.
+         * @param {string} description Error message to add to assert in case of
+         *                             failure.
+         * @param {number} timeout Timeout in ms. This is multiplied by the global
+         *                         timeout_multiplier
+         * @param {number} interval Polling interval in ms
+         *
+         **/
+        var timeout_full = timeout * tests.timeout_multiplier;
```
Comment (annevk) [inline resources/testharness.js:2062]: Let's use const / let for new things?

Comment (jgraham) [reply resources/testharness.js:2062]: Historically we've been very conservative using "new" js features in `testharness.js`. idk if these are now new enough that we can just use them. I suspect the answer is "yes", but I'm not sure I want to make that decision unilaterally.

**Comment by annevk** [comment]
Do not merge before https://github.com/web-platform-tests/rfcs/pull/56 has landed.

**Review by annevk — APPROVED** [review]

**Pull request merged**